src/Processors/QueryPlan/JoinStepLogical.cpp

```diff
@@ -435,14 +435,14 @@ JoinActionRef buildSingleActionForJoinCondition(const JoinCondition & join_condi
     auto left_filter_conditions_action = concatConditions(join_condition.left_filter_conditions, expression_actions.left_pre_join_actions, query_context);
     if (left_filter_conditions_action)
     {
-        left_filter_conditions_action.node = &findInput(expression_actions.post_join_actions, left_filter_conditions_action);
+        left_filter_conditions_action.node = &findInput(expression_actions.post_join_actions, left_filter_conditions_action, true);
```

Comment:
Could you please explain why we have to allow adding expressions? I cannot figure it out.